spyder/api/plugins.py

```diff
@@ -533,3 +557,874 @@ def set_ancestor(self, ancestor):
+        -----
+        This is a method to be able to update localization without a restart
+        """
+        raise NotImplementedError('A plugin name must be defined!')
+
+    def get_description(self):
+        """
+        Return the plugin localized description.
+
+        Returns
+        -------
+        str
+            Localized description of the plugin.
+
+        Notes
+        -----
+        This is a method to be able to update localization without a restart
+        """
+        raise NotImplementedError('A plugin description must be defined!')
+
+    def get_icon(self):
+        """
+        Return the plugin associated icon.
+
+        Returns
+        -------
+        QIcon
+            QIcon instance
+        """
+        raise NotImplementedError('A plugin icon must be defined!')
+
+    # --- API: Optional methods to override ----------------------------------
+    # ------------------------------------------------------------------------
+    def register(self):  # Old name: register_plugin
+        """
+        Setup and register plugin in Spyder's main window and connect it to
+        other plugins.
+        """
+        pass
+
+    def unregister(self):
+        """
+        Disconnect signals and clean up the plugin to be able to stop a plugin
+        while Spyder is running.
+        """
+        pass
+
+    @staticmethod
+    def check_compatibility():
+        """
+        This method can be reimplemented to check compatibility of a plugin
+        with the user's current environment.
+
+        Returns
+        -------
+        (bool, str)
+            The first value tells Spyder if the plugin has passed the
+            compatibility test defined in this method. The second value
+            is a message that must explain users why the plugin was
+            found to be incompatible (e.g. 'This plugin does not work
+            with PyQt4'). It will be shown at startup in a QMessageBox.
+        """
+        valid = True
+        message = ''  # Note: Remeber to use _('') to localize the string
+        return valid, message
+
+    def on_first_registration(self):
+        """
+        Action to be performed on the first time the plugin is started.
+
+        Or to perform actions that are needed only the first time this is
+        installed.
+
+        This method is called after the main window is visible.
+        """
+        pass
+
+    # Old name: closing_plugin
+    def on_close(self, cancelable=False):
+        """
+        Perform actions before the main window is closed.
+
+        Returns
+        -------
+        bool
+            Whether the plugin may be closed immediately or not.
+
+        Notes
+        -----
+        The returned value is ignored if *cancelable* is False.
+        """
+        return True
+
+    def update_font(self):
+        """
+        This must be reimplemented by plugins that need to adjust their fonts.
+        """
+        pass
+
+    def update_style(self):
```

Comment:
What this refers to? Please clarify it in the docstrings below, along with an example.

Reply:
Fixed